src/file_auth.ts

```diff
@@ -0,0 +1,48 @@
+import fs = require('fs');
+import https = require('https');
+import request = require('request');
+
+import { Authenticator } from './auth';
+import { User } from './config_types';
+
+export class FileAuth implements Authenticator {
+    private token: string | null = null;
+    private lastRead: Date | null = null;
+
+    public isAuthProvider(user: User): boolean {
+        return user.authProvider && user.authProvider.config && user.authProvider.config.tokenFile;
+    }
+
+    public getToken(user: User): string | null {
+        if (this.token == null) {
+            this.readToken(user.authProvider.config.tokenFile);
+        }
+        if (this.isTokenExpired()) {
+            this.readToken(user.authProvider.config.tokenFile);
+        }
+        return this.token;
+    }
+
+    public async applyAuthentication(user: User, opts: request.Options | https.RequestOptions) {
+        // nothing to do.
+    }
+
+    private readToken(filePath: string) {
```

Comment:
renaming nit: `readToken` to `updateToken`

I'd expect `readToken` to return something read.  `updateToken` implies updating some state.

Reply:
`refreshToken` feels like a better name for this.